xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -1552,24 +1552,16 @@ public Date getResourceLastModificationDate(String name)
 
     public byte[] getResourceContentAsBytes(String name) throws IOException
     {
-        InputStream is = null;
+        byte output[] = new byte[0];
         if (getEngineContext() != null) {
-
-            try {
-                is = getResourceAsStream(name);
-            } catch (Exception e) {
+            try (InputStream is = getResourceAsStream(name)) {
+                if (is == null) {
+                    output = FileUtils.readFileToByteArray(new File(name));
+                } else
+                    output = IOUtils.toByteArray(is);
             }
         }
-
-        if (is == null) {
-            return FileUtils.readFileToByteArray(new File(name));
-        }
-
-        try {
-            return IOUtils.toByteArray(is);
-        } finally {
-            IOUtils.closeQuietly(is);
-        }
+        return output;
```

Comment:
@tmortagne is it okay to return an empty byte array if the engine context is null?

Reply:
or maybe throw an exception?

Reply:
Actually I mixed with the other pull request, there was indeed a catch before so you should put it back, sorry.